### PR TITLE
Fixes from IBOIS's dry run

### PR DESCRIPTION
**Pull request by 9and3**
## description
This PR introduces small fixes while testing the subtractive chapter and pipeline from IBOIS's side.

## fixes
- no more error for the `DF_build_assembly`, `DF_cloud_voxel_downsample`, `DF_cloud_size_downsample`, `DF_cloud_normal_estimator` and many others as dropped on the canvas
- the assembly preview is been fixed and it is now working for single beams
- added the `mean` value to all the error compute, visualizer and csv exporter

### CODE REVIEW

File: src/gh/components/DF_cloud_mesh_distance/code.py [hunk 1]
```diff
@@ -43,7 +41,11 @@ def RunScript(self,
             rh_mesh_target_list = [joint.to_mesh(i_analysis_resolution) for joint in i_assembly._all_joints]
         elif len(i_assembly.all_joint_faces) == len(i_cloud_list):
             ghenv.Component.Message = "Per Joint Face"  # noqa: F821
-            rh_mesh_target_list = [joint_face.to_mesh() for joint_face in i_assembly.all_joint_faces]
+            rh_mesh_target_list = []
+            for joint in i_assembly.all_joints:
+                for face in joint.faces:
+                    face = face.to_mesh()
```
Comment (eleniv3d) [inline src/gh/components/DF_cloud_mesh_distance/code.py:47]: Small comment here and perhaps not directly related to the PR.. for the case of beam or joint we are using the `joint.to_mesh()` and `beam.to_mesh()` that get the input of the `MaximumEdgeLength` which the user directly changes when meshing the geometry. This is not the case now for the face of each joint where `Rhino.Geometry.MeshingParameters.QualityRenderMesh` is always used.

**Comment by 9and3** [comment]
> I believe Normal Segmentator and Colorize Cloud are still red when dropped on the canvas, otherwise no complaints.
Would you be able to add a commit to fix this with the `if..` catch at the beginning plis? Thanks 👐 

btw @Eleni I am almost finished with the refactored that you might want to test, I still have it for a couple of hours..I'll pop a message here!

**Comment by eleniv3d** [comment]
> > I believe Normal Segmentator and Colorize Cloud are still red when dropped on the canvas, otherwise no complaints.
> 
> Would you be able to add a commit to fix this with the `if..` catch at the beginning plis? Thanks 👐
> 
> btw @eleni I am almost finished with the refactored that you might want to test, I still have it for a couple of hours..I'll pop a message here!

Okie. Done ✔️

In case the checks fail me, let the record show that I tried..

Screenshot 2024-10-03 154457

**Comment by 9and3** [comment]
> In case the checks fail me, let the record show that I tried..

the hereby CI jury has emitted a "not guilty" output

image

**Comment by 9and3** [comment]
@eleniv3d mesh swap is also fine!

**Pull request merged**